### ISSUE TITLE
Custom radix

### ISSUE THREAD

**Issue by zjeff222** [KenKundert/quantiphy#8]
Hi,
This looks much better than engfmt, thanks for the tip!  Among the new stuff I want to use, I found the radix preference.  Is it for the input, output or both?  

from the doc: 
radix (str) – The character to be used as the radix. By default it is ‘.’.
radix – The text to be used as the plus sign. By default it is ‘+’, but is sometimes ‘＋’ (the unicode full width plus sign) or ‘’ to simply eliminate plus signs from numbers. You can access the Unicode full width plus sign using Quantity.plus_sign.

First, I assume the second radix is an error and should read 'plus' or something like that.

So for the actual radix, it does not seam to work for the output:

```
import quantiphy as eng
eng.Quantity.set_prefs(radix=',')
t= eng.Quantity(3.123e-6,'m')
print(t.render())
```
results in 3.123 um

as for the input, a comma will be interpreted as a 3 digit separator (3,123e-6 m -> 3.123 mm)
thanks

**Comment by KenKundert** [comment]
Oh no! That is quite a blunder.  I'll get to work on fixing them.

**Comment by KenKundert** [comment]
Okay, I have fixed the version in github. I'll probably update pypi in the next few days. Is it possible for you to check out the github version and try it out. You can let me know if you notice anything else I forgot.

**Comment by zjeff222** [comment]
Thanks for the fix.  
The output is now working as expected, but the input is not.
`from quantiphy import Quantity
Quantity.set_prefs(radix=',')
t= Quantity('3,123','m')
print(t.render())`

yields 3,123 km 

so we do have a comma in the output as wanted but since the quantity is now in km, it shows that the input comma was interpreted as a thousand seperator and not a radix.  Maybe I was not clear in the last message - in comma radix systems, the comma is used exclusively as a radix.  So 3,123 m is equivalent to 3.123 m

**Comment by KenKundert** [comment]
Actually, I had not implemented radix,comma support on the input; only on the output. However, I have updated it again and now the latest version should work. Please try the new version: 2.8.4. But you will need to set both radix and comma. By only setting radix, you still had ',' as the comma, and commas are stripped from the number.

**Comment by KenKundert** [comment]
I have updated again. It is now an error for radix and comma to be the same.
```
>>> from quantiphy import Quantity, QuantiPhyError                               
                                                                                 
>>> try:                                                                         
...     with Quantity.prefs(radix=','):                                          
...         q = Quantity('3,123','m')                                            
...         print(q.render())                                                    
... except QuantiPhyError as e:                                                  
...     print(e.render())                                                        
comma and radix must differ.                                                                   
                                                                                 
>>> try:                                                                         
...     with Quantity.prefs(radix=',', comma='.'):                               
...         q = Quantity('3,123','m')                                            
...         print(q.render())                                                    
... except QuantiPhyError as e:                                                  
...     print(e.render())                                                        
3,123 m                                                                          
```
Version is now 2.8.5.

**Comment by zjeff222** [comment]
Great.  I did limited testing and now it works as expected for comma radix standard. Also, I noticed that when input is set with radix=',', comma=' ' (official/ISO standard in my region), the input with a dot will still work - and I think it is a good thing because not everyone uses the local standard.

thanks again

**Comment by KenKundert** [comment]
Thank you. I really appreciate you reporting these issues.
I have updated the documentation and the version, it is no 2.8.6. The additions to the documentation can be found [here](https://quantiphy.readthedocs.io/en/latest/user.html#localization). I will try to update pypi to in the next few days.

**Comment by KenKundert** [comment]
I have released the fixed version (vers 2.9) on pypi.
